lib/checkother.cpp

```diff
@@ -3685,6 +3685,10 @@ void CheckOther::checkComparePointers()
                 continue;
             if (var1->isRValueReference() || var2->isRValueReference())
                 continue;
+            if (var1->scope() && var1->scope()->isClassOrStructOrUnion() && var1->scope()->isNestedIn(var2->typeScope()))
+                continue;
+            if (var2->scope() && var2->scope()->isClassOrStructOrUnion() && var2->scope()->isNestedIn(var1->typeScope()))
+                continue;
```

Comment:
Wont this skip `&s1.i - (int*)&s2`? I think we have a `getParentLifetime` function that would help here.